CustomQueries/server/aspnet/RevealSdk.Server/Reveal/DataSourceProvider.cs

```diff
@@ -12,8 +12,8 @@ public class DataSourceProvider : IRVDataSourceProvider
         {
             if (dataSourceItem is RVSqlServerDataSourceItem sqlDataSourceItem)
             {
-                var sqlDataSource = (RVSqlServerDataSource)sqlDataSourceItem.DataSource;
-                UpdateDataSource(sqlDataSource);
+                //update underlying data source
+                ChangeDataSourceAsync(userContext, sqlDataSourceItem.DataSource);
```

Comment:
Can't this be called once as the first line of code in the change data source item async method? I'm pretty sure the base data source item has the data source property, so we do not need to duplicate this call for each data source.